engine/runtime/src/main/scala/org/enso/compiler/pass/resolve/GenerateDocumentation.scala

```diff
@@ -0,0 +1,192 @@
+package org.enso.compiler.pass.resolve
+
+import org.enso.compiler.context.{InlineContext, ModuleContext}
+import org.enso.compiler.core.IR
+import org.enso.compiler.core.IR.Case.Branch
+import org.enso.compiler.core.ir.MetadataStorage._
+import org.enso.compiler.pass.IRPass
+import org.enso.compiler.pass.resolve.DocumentationComments.Doc
+import org.enso.docs.generator.DocParserWrapper
+
+/** Generates documentation on resolved IR.
+  */
+case object GenerateDocumentation extends IRPass {
+
+  /** The type of the metadata object that the pass writes to the IR. */
+  override type Metadata = DocumentationComments.Doc
+
+  /** The type of configuration for the pass. */
+  override type Config = IRPass.Configuration.Default
+
+  /** The passes that this pass depends _directly_ on to run. */
+  override val precursorPasses: Seq[IRPass] = Seq(DocumentationComments)
+
+  /** The passes that are invalidated by running this pass. */
+  override val invalidatedPasses: Seq[IRPass] = Seq()
+
+  /** Collects comments for a module and assigns them to the commented
+    * entities as metadata.
+    *
+    * @param ir the Enso IR to process
+    * @param moduleContext a context object that contains the information needed
+    *                      to process a module
+    * @return `ir`, possibly having made transformations or annotations to that
+    *         IR.
+    */
+  override def runModule(
```

Comment:
Two things wrong with this pass:
1. It goes into method bodies, even though you're only interested in toplevel docs.
2. It is a copy paste of `DocComments`, which is pointless at this point. `DocumentationComments` already reattaches (and removes) doc nodes from the AST. So what you want to do is: 
– Make sure this pass runs after DocumentationComments. Give it an explicit dependency too.
– Iterate over the toplevel bindings and just read the `DocumentationComments` metadata from each node. You don't need to think about the original IR structure anymore, the job of DocumentationComments is to translate that into a consumable form.